### ISSUE TITLE
Displaying images from network as marker clusters

### ISSUE THREAD

**Issue by ProgUser145** [mapbox/mapbox-maps-flutter#780]
Hi, thanks for this awesome package.

I’m working on a feature where I need to handle marker clusters dynamically, and I have a specific use case:

- When there are multiple items in the same area, they should cluster as expected.

- When zooming in and the cluster count reduces to 1 (essentially resolving into a single marker), I want to dynamically display an image fetched from network (e.g., an AWS presigned URL). So basically call an API which comes with a response as url.
 
- Each marker should display a different image based on the API response(similar to how Instagram displays stories).

I understand that it’s possible to display the same image for all markers, but is it feasible to assign different images to each marker dynamically in this way? In flutter_map this is possible since the markers/clusters are widgets, but would love to know if it is possible to do it also with this package

<img src="https://github.com/user-attachments/assets/88adce54-a847-45d3-8400-1851dc7c003d" width=30% height=30%>

**Comment by UsamaEjaz0** [comment]
Did you find a solution?

**Comment by BogibekDev** [comment]
@ProgUser145 @UsamaEjaz0   Did you find a solution?

**Comment by O-Hannonen** [comment]
Sorry for the super late response. You've probably already solved this, but for others possibly stumbling on this issue, here is some direction to start looking into.

> In flutter_map this is possible since the markers/clusters are widgets, but would love to know if it is possible to do it also with this package

In case something similar to this is what you are looking for, this would be possible once #53 lands.

Meanwhile (and possibly after View Annotations land too, depending on you exact use-case), you can work around the issue. Since each marker needs a genuinely unique image, a bounded icon-variant set won't cover this the way it can for a fixed set of categories, but you still don't need Flutter widget capture to solve it. So either one of these options could work for you:

1. fetch/decode each image once, then register it with `addStyleImage` under a per-item id (e.g. keyed by the item's id or its URL). If you back your markers with a `GeoJsonSource` + `SymbolLayer` (rather than `PointAnnotationManager`), you can then reference those ids with a data-driven icon-image expression (eg. `"icon-image": ["get", "imageId"]`) where each feature carries its own `imageId` property pointing at the id you registered for that marker's photo. That keeps image compositing native instead of round-tripping through Flutter's renderer per marker, and it scales fine as long as the number of distinct images on screen at once is reasonable (each one is a real registered style image).

2. If you're using `PointAnnotationManager` instead of a raw `SymbolLayer`, the same idea still works but isn't expression-driven. Annotations take a literal image id, not an expression, so you'd `addStyleImage` each fetched photo once and then set image to that id directly on each `PointAnnotationOptions` / `PointAnnotation` when creating or updating it.

Closing as duplicate of #53 to track the progress and discussion there.
